feat(component): add description to select option and select action

golcinho opened the pull request:
## What?
Adding a description to the `select option` and `select action`.

## Screenshots
<img width="453" alt="Screenshot 2020-07-28 at 11 15 05" src="https://user-images.githubusercontent.com/9980803/88638053-a2fe5b00-d0c3-11ea-8131-aad6bec6aed0.png">
<img width="453" alt="Screenshot 2020-07-30 at 09 45 23" src="https://user-images.githubusercontent.com/9980803/88890114-d10da780-d249-11ea-86b7-03d57c163e53.png">
<img width="431" alt="Screenshot 2020-07-30 at 09 45 38" src="https://user-images.githubusercontent.com/9980803/88890118-d2d76b00-d249-11ea-81da-13c749a5105a.png">
<img width="967" alt="Screenshot 2020-07-28 at 09 38 44" src="https://user-images.githubusercontent.com/9980803/88629491-ff5b7d80-d0b7-11ea-80cf-fe3c56f0929c.png">
<img width="969" alt="Screenshot 2020-07-28 at 09 39 01" src="https://user-images.githubusercontent.com/9980803/88629495-008caa80-d0b8-11ea-80e3-ce98e0d9b9e9.png">

## Testing
Unit tests.

valterfatia commented:
Nice work! Two small things from me:
1. The description text should be in "secondary60" and "small text" – or was there a different direction from another designer?
2. I'm unsure about changing the colour of the description when there's an interaction – currently reviewing within the design team, hoping to have it decided soon ;)

golcinho commented:
> Nice work! Two small things from me:
> 
> 1. The description text should be in "secondary60" and "small text" – or was there a different direction from another designer?
> 2. I'm unsure about changing the colour of the description when there's an interaction – currently reviewing within the design team, hoping to have it decided soon ;)

Hey @valterfatia !

Thank you for the comments!

I changed color to `secondary60` and description text is `Small` (14px).

Thank you for your assistance with the second point 🤝

golcinho commented:
> Everything looks good, but do you think we can add these changes to the `MultiSelect` component as well?
> 
> Thanks for adding this! 👍

Good point, thank you 🙇 

I will make it for `MultiSelect` too 👍

chanceaclark commented on packages/docs/pages/Select/SelectPage.tsx:
```diff
@@ -342,6 +342,53 @@ const SelectPage = () => (
+
+    <H2>Option & Action Description</H2>
+
+    <Text>It is possible to add a description for select options and select action</Text>
```
```suggestion
    <Text>It is possible to add a description for select options and actions</Text>
```

chanceaclark commented on packages/big-design/src/components/Select/Select.tsx:
```diff
@@ -465,3 +480,15 @@ const iconColor = <T extends any>(item: SelectOption<T> | SelectAction, isHighli
+    return 'secondary60';
+  }
+
+  return 'actionType' in item ? (item.actionType === 'destructive' ? 'danger30' : 'primary30') : 'primary30';
```
@valterfatia `danger30` color with `danger10` background-color doesn't meet accessibility color contrast ratios. Is there another option?

valterfatia replied:
I have been reviewing that change within the design team and we have agreed to keep secondary60 as the text colour to meet accessibility requirements.

valterfatia commented:
Clarifying the last point, from my previous comment lets keep the text as secondary60 on hover and just do the background change as we already have.

valterfatia commented:
Okay I have one more request – which is likely better addressed in a separate PR/issue – but I wanted to call it out here.

All text in a dropdown should remain left aligned to a single "key line" regardless of the presence of icons within the dropdown. In practice this when an item in a dropdown has an icon, then other elements without an icon should keep the same gap so the text aligns in a single line.

Ideally we should aim for having dropdowns items should all have a consistent look and feel, but we also accept in some instances this won't be possible. In those cases our request is for the space to be left empty.

chanceaclark commented:
> Okay I have one more request – which is likely better addressed in a separate PR/issue – but I wanted to call it out here.
> 
> All text in a dropdown should remain left aligned to a single "key line" regardless of the presence of icons within the dropdown. In practice this when an item in a dropdown has an icon, then other elements without an icon should keep the same gap so the text aligns in a single line.
> 
> Ideally we should aim for having dropdowns items should all have a consistent look and feel, but we also accept in some instances this won't be possible. In those cases our request is for the space to be left empty.

Let's talk about this offline in our Feedback Session.

valterfatia commented:
> Let's talk about this offline in our Feedback Session.

Sounds good!

golcinho commented:
@chanceaclark @valterfatia I changed description color to `secondary60` for the select options and actions. (screenshots are updated).

Also, I added a description of the options and actions for the `MultiSelect` component.

cc @jorgemoya

chanceaclark commented on packages/docs/pages/MultiSelect/MultiSelectPage.tsx:
```diff
@@ -300,6 +300,53 @@ const MultiSelectPage = () => (
+
+    <H2>Option & Action Description</H2>
+
+    <Text>It is possible to add a description for select options and select action</Text>
```
```suggestion
    <Text>It is possible to add a description for select options and actions.</Text>
```

chanceaclark commented on packages/docs/pages/Select/SelectPage.tsx:
```diff
@@ -342,6 +342,53 @@ const SelectPage = () => (
+
+    <H2>Option & Action Description</H2>
+
+    <Text>It is possible to add a description for select options and actions</Text>
```
```suggestion
    <Text>It is possible to add a description for select options and actions.</Text>
```
